### PR TITLE
boards: nrf_bsim: enable verbose log options

### PR DESCRIPTION
**Pull request by jori-nordic**
Function and thread names are great for debugging.

### CODE REVIEW

**Review by aescolar — COMMENTED** [review]
The logger multidomain test has the same issue.
Maybe do a whole twister run locally for this board

File: boards/posix/nrf_bsim/Kconfig.defconfig [hunk 1]
```diff
@@ -81,6 +81,26 @@ choice LOG_MODE
 	default LOG_MODE_IMMEDIATE
 endchoice
 
+# Since flash & ram is "free", we can also include the function names
+# in all logging levels.
+config LOG_FUNC_NAME_PREFIX_ERR
```
Comment (aescolar) [inline boards/posix/nrf_bsim/Kconfig.defconfig:86]: I'm ok with this changes, but others who use this target quite a bit may want to chip in.
@Emil-Gydesen-Bose @kruithofa @cvinayak @alxelax @PavelVPV

Comment (jori-nordic) [reply boards/posix/nrf_bsim/Kconfig.defconfig:86]: I don't think it does. it's only enabled for LOG_DBG like the rest of zephyr.
This also adds the thread name which is super useful when you have multiple contexts running the same fns

**Review by PavelVPV — APPROVED** [review]

**Comment by jori-nordic** [comment]
Ran twister locally, too many test failures, I ain't got time fo this. Will just use overlay + env var instead.

**Pull request closed without merging**